apps/core/app/api/cart-quantity/route.ts

```diff
@@ -0,0 +1,19 @@
+import { cookies } from 'next/headers';
+import { NextResponse } from 'next/server';
+
+import { getCart } from '~/client/queries/get-cart';
+
+export const GET = async () => {
+  const cartId = cookies().get('cartId')?.value;
+  let count = 0;
+
+  if (cartId) {
+    const cart = await getCart(cartId);
+
+    count = cart?.lineItems.totalQuantity ?? 0;
+  }
+
+  return NextResponse.json({ count });
+};
```

Comment:
We try to avoid using `let`'s and mutations.
```suggestion
export const GET = async () => {
  const cartId = cookies().get('cartId')?.value;

  if (cartId) {
    const cart = await getCart(cartId);

    return NextResponse.json({ count: cart?.lineItems.totalQuantity ?? 0 });
  }

  return NextResponse.json({ count: 0 });
};
```